File: 66-plus-one/plus-one.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        n = len(digits)
        carry = 1

        for i in range(n - 1, -1, -1):
            total = digits[i] + carry
            digits[i] = total % 10
            carry = total // 10

        # 만약 마지막까지 전파된 carry가 있다면, 배열을 확장하고 맨 앞에 carry를 추가합니다.
        if carry > 0:
            digits.insert(0, carry)

        return digits

File: 66-plus-one/plus-one.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* plusOne(int* digits, int digitsSize, int* returnSize) {
    int carry = 1;

    for (int i = digitsSize - 1; i >= 0; i--) {
        int sum = digits[i] + carry;
        digits[i] = sum % 10;
        carry = sum / 10;
    }

    if (carry > 0) {
        *returnSize = digitsSize + 1;
        int* result = (int*)malloc(sizeof(int) * (*returnSize));
        result[0] = carry;
        for (int i = 1; i < *returnSize; i++) {
            result[i] = digits[i - 1];
        }
        return result;
    } else {
        *returnSize = digitsSize;
        return digits;
    }
}